code/game/objects/items/weapons/lighters.dm

```diff
@@ -33,54 +23,46 @@
 	icon_state = icon_off
 
 /obj/item/lighter/attack_self(mob/living/user)
-	if(user.r_hand == src || user.l_hand == src || isrobot(user))
-		if(!lit)
-			lit = 1
-			w_class = WEIGHT_CLASS_BULKY
-			icon_state = icon_on
-			item_state = icon_on
-			force = 5
-			damtype = "fire"
-			hitsound = 'sound/items/welder.ogg'
-			attack_verb = list("burnt", "singed")
-			if(istype(src, /obj/item/lighter/zippo) )
-				user.visible_message("<span class='rose'>Without even breaking stride, [user] flips open and lights [src] in one smooth movement.</span>")
-				playsound(src.loc, 'sound/items/zippolight.ogg', 25, 1)
-			else
-				if(prob(75))
-					user.visible_message("<span class='notice'>After a few attempts, [user] manages to light the [src].</span>")
-				else
-					to_chat(user, "<span class='warning'>You burn yourself while lighting the lighter.</span>")
-					var/mob/living/M = user
-					if(ishuman(M))
-						var/mob/living/carbon/human/H = M
-						var/obj/item/organ/external/affecting = H.get_organ("[user.hand ? "l" : "r" ]_hand")
-						if(affecting.receive_damage( 0, 5 ))		//INFERNO
-							H.UpdateDamageIcon()
-					user.visible_message("<span class='notice'>After a few attempts, [user] manages to light the [src], [user.p_they()] however burn[user.p_s()] [user.p_their()] finger in the process.</span>")
-
-			set_light(2)
-			START_PROCESSING(SSobj, src)
-		else
-			lit = 0
-			w_class = WEIGHT_CLASS_TINY
-			icon_state = icon_off
-			item_state = icon_off
-			hitsound = "swing_hit"
-			force = 0
-			attack_verb = null //human_defense.dm takes care of it
-			if(istype(src, /obj/item/lighter/zippo) )
-				user.visible_message("<span class='rose'>You hear a quiet click, as [user] shuts off [src] without even looking at what [user.p_theyre()] doing. Wow.")
-				playsound(src.loc, 'sound/items/zippoclose.ogg', 25, 1)
-			else
-				user.visible_message("<span class='notice'>[user] quietly shuts off the [src].")
-
-			set_light(0)
-			STOP_PROCESSING(SSobj, src)
+	. = ..()
+	if(!lit)
+		turn_on_lighter(user)
 	else
-		return ..()
-	return
+		turn_off_lighter(user)
+
+/obj/item/lighter/proc/turn_on_lighter(mob/living/user)
+	lit = TRUE
+	w_class = WEIGHT_CLASS_BULKY
+	icon_state = icon_on
+	item_state = icon_on
+	force = 5
+	damtype = BURN
+	hitsound = 'sound/items/welder.ogg'
+	attack_verb = list("burnt", "singed")
+
+	if(prob(75) || issilicon(user)) // Robots can never fail to light it.
+		to_chat(user, "<span class='notice'>You light [src].</span>")
+	else
+		var/mob/living/carbon/human/H = user
+		var/obj/item/organ/external/affecting = H.get_organ("[user.hand ? "l" : "r" ]_hand")
+		if(affecting.receive_damage( 0, 5 ))		//INFERNO
+			H.UpdateDamageIcon()
+		to_chat(user,"<span class='notice'>You light [src], but you burn your hand in the process.</span>")
+
+	set_light(2)
+	START_PROCESSING(SSobj, src)
```

Comment:
Note: The way this is written, zippo lighters will now have a 25% burn chance like regular lighters, so you might want to refactor this bit, perhaps splitting it into `attempt_light` or something

Reply:
Oops yeah zippo's are not meant to have that chance.

Reply:
Would giving lighters a `light_chance` var and then doning `if(prob(light_chance)..` be acceptable?

Reply:
That ought to work, although you might also want a variable for the message to display to others, since the zippo has a unique one

Reply:
Yeah I still display that in the `turn_on_lighter` override for the zippos.

Reply:
I'm mostly concerned about it displaying multiple messages instead of just the one - that seems a bit more janky

Reply:
Fair, it is janky. I'll do what you said earlier, throw it into a seperate proc, and make it so if the fancy message gets shown it doesn't display the other one